### ISSUE TITLE
identifiable resource string

### ISSUE THREAD

**Issue by mightyBroccoli** [tigase/siskin-im#69]
Would it be possible to set the resouce string to something groupable eg. Siskin.UIDevice.current.name ?
This would be nice to monitor the connected users, based on their resource string.

https://github.com/tigase/siskin-im/blob/209812c20c1131e53b2fb5cdccf2995d64ff2c1f/SiskinIM/service/XmppService.swift#L569

**Comment by hantu85** [comment]
It could be done, but I wonder why it should be? Resource is an opaque value and should not be considered to have any meaningful details in it.

Why I understand "data gathering" (statistics) purposes I'm not sure that this is how it should be done.

**Comment by mightyBroccoli** [comment]
It is true that the resource shound't have any meaningful details in them, but they should be unique across all user connections. I noticed that my two test ios devices both register "iphone" as their resource which results in them being kicked constantly due to resource conflicts.
You are correct in the sense that resource strings aren't purposed to be identifiable. I wasn't asking to put some valueable info into them rather some recognisable pattern. Something like Conversations ( Conversations.6uUn ) or Gajim ( Gajim.28as3nh3 ) or Dino ( dino.2syds ) and so on.
I think that these random letters aren't considered meanigful, but they are recognisable which is a benefit.

**Comment by hantu85** [comment]
SiskinIM uses as a resource device name, so if you had two phones named "iphone" that caused the issue.

**Comment by mightyBroccoli** [comment]
yes I knew that already.
I assume that you are not willing to include a regocnisable pattern into the client resource right?

**Comment by hantu85** [comment]
I'm not convinced that this is required or needed but may appear in the future. We will think about it for future releases.

**Comment by Neustradamus** [comment]
@hantu85: Can you add a feature to select a ressource name instead of the automatic phone name?

You can see in Psi/Psi+
-> Psi -> Options -> Accounts -> Account Properties -> Misc.
Resource selection:
- Manual and a form
- Use host name

**Comment by hantu85** [comment]
Siskin is for a phone not for a desktop. I'm not willing to add manual selection of a resource. Its configuration is already too complex for many users so adding anything new to configure is just a bad idea.

**Comment by hantu85** [comment]
And to change a "resource" you can always change your phone name.

**Comment by Neustradamus** [comment]
It is same for Desktop and Phone, of course it is to all Tigase apps:
- Desktop apps: "Manual" or "Use host name"
- Phone apps: "Manual" or "Use phone name"

Note: It is possible to have not rights to change the phone name.

**Comment by hantu85** [comment]
I disagree, phone app and its features are different from the desktop app and other things are required. 

To sum up - feature to change "resource" will not be added.

**Comment by DanScharon** [comment]
> And to change a "resource" you can always change your phone name.

this seems to have no effect. Even after rebooting the phone or reinstalling the app. Any advice?
